### ISSUE TITLE
Adult Website Fork Ideas...

### ISSUE THREAD

**Issue by darvell** [sherlock-project/sherlock#854]
- [x] I'm asking a question regarding Sherlock
- [x] My question is not a tech support question.

## Question

It's been mentioned that sherlock does not want adult website support in it for understandable reasons but a lot of OSINT work does require checking against them. For a lot of these websites we need the ability to create classes for specific websites or expand the data.json structure to include ways to allow authentication.

The idea of allowing credentials to check for other users on websites that do require this something that could be core to sherlock, so the idea of having a package called 'auth' which can have per-website classes that follow the same interface to handle authenticating the session correctly, reading info from something like ~/.sherlock/credentials.json and if credentials exist, the website is checked.

The second part of this is holding an 'adult' version of this that would allow for external modules and an external data.json to be merged together. If considered community maintained, would it be possible to have a `sherlock-project/veronica-mars` (pick your own infidelity private investigator) repo to have those and supplying the flag --adult to sherlock would pull data from that repo on top for that run? That way we do not need to be holding forks all over the place, just keeping things separate.

**Comment by darvell** [comment]
If it would help, I'd be happy to lead on this there are not many websites I can think of but, it doesn't hurt to add some of the odder parts of the internet.

**Comment by GeneralDenmark** [comment]
I like the idea that @darvell suggests, as this allows for adult sites to also be contributed to this repo, without the hassel of forks. And also allows the users that does not want adult sites, to avoid getting matches, by not using the parameter.

**Comment by hoadlck** [comment]
You can already pass other data.json files on the command line without any alteration of Sherlock.  So, there is no need for forks:  sites that we don't want to have in the core Sherlock can be in there own repository, and users can pass them in.

I imagine that some config file or environment variable check could be added to Sherlock so that users were not forced to add the data.json to every command link query.

**Comment by darvell** [comment]
Sure, the reason I said a fork is because looking at other times it has been mentioned, it's been requested to keep explicitly out of this repository. There already is starting to become some bloat with data.json being hard to parse with websites lacking context as to what category they are among other things, I think that there is a better approach to all of this that not only allows for adult sites if the user chooses but to be able to filter down categories of websites by introducing just another field.

It's something I've just been thinking on for a while, mixed with the idea of having a very basic web GUI to modify/test data.json and add entries, categorize them, etc. to allow for easier collaboration and management. That's a bit off-topic though. But, having maybe sherlock-project/sherlock-data-files and holding the respective data.json's in there mixed with being able to say e.g. --category social,forums,adult and pulling their respective data.json's from there would be nice. I'd fork it and do it myself but it seems pointless if it's not an idea that the project author would want.

**Comment by hoadlck** [comment]
Yes, we have talked about adding topic support to allow people to limit the scope of their search.  Somewhere around here there is a proposal:  I think it was something like `--topic social --topic video` for a search for social video sites.  Maybe `--category` is more descriptive.

Ultimately, the data should be in a database:  including it in Git (especially in the same repository as the algorithmic code) causes much unneeded churn.

But, the highest priority for me is to get Sherlock packaged like a normal Python module.  This will get rid of most of the problems users have when trying to use Sherlock.

**Comment by darvell** [comment]
Understandable, I actually really wanted to start work on that so it can be integrated in to other OSINT tools if needed, is changing Sherlock to a module something you'd be interested in getting outside help on?

**Comment by hoadlck** [comment]
Sherlock is open source, so there is no "outside help". :) 

The wiki has my current notes on the [restructure](https://github.com/sherlock-project/sherlock/wiki/Restructure-Notes).  The biggest remaining task is changing the way that the actual queries are run (see Detection Method Abstraction, and Detection Engine).  The current module is not structured in a way that will be useful for people integrating it into other tools.  And, we want to be able to add other detection methods without having to expand the same function.  The current core function is all one big chunk.

**Comment by sdushantha** [comment]
I am currently developing a newer version of Sherlock that will be modular so that it can be available on PyPi, have an Python and web API and possibly a web interface. It will allow for categories/tags. Therefore I will be closing this as this is not something I will be doing with the current version of Sherlock
